### DISCUSSION TITLE
better UX for providing info to user when requesting input

### DISCUSSION THREAD

**Discussion by msorens** [microsoft/vscode-discussions#435] [Extension Development QnA] [answered]
Since a VSCode user may deliberately—or inadvertently!—turn off auto-scrolling in the output pane, I am wondering if one can programmatically scroll the output pane to the bottom? I have been unable to find any way to do this from the API docs, though... hoping there might be a trick to do this somehow.

The reason I need this is that I am prompting the user with a `QuickPick` but I want to give the user more than just a few words of description to guide them. So... another possibility: is there any way to embed newlines into the `prompt` ? I would particularly like to add a line break at the end of my prompt string—as shown with the arrow—but more generally would like to be able to add line breaks anywhere in the prompt string. I have tried a literal `\n` as well as using a template literal with actual line breaks in it, but neither of those worked.

<img width="679" alt="image" src="https://user-images.githubusercontent.com/6817500/216414070-88dd876b-690e-4774-8dc3-13dcbdf0acf2.png">

**Comment by msorens** [comment]
Other things I have tried out of desperation:

These all vanish in the rendered HTML:
- Single embedded newline: `'foo\nbar'`
- Double embedded newline:`'foo\n\nbar'`
- Single embedded return/newline: `'foo\r\nbar'`
- Template literal with actual return: \`foo
bar\`

These are all rendered literally, not interpreted.
- Single embedded return/newline as entities: `'foo&#13;&#10;bar'`
- HTML element: `'foo<br>bar'`

Since the [docs](https://code.visualstudio.com/api/extension-capabilities/overview#no-dom-access) clearly state "Extensions have no access to the DOM of VS Code UI. You cannot write an extension that applies custom CSS to VS Code or adds an HTML element to VS Code UI." I see no way of adding any kind of hook for this, so looking rather bleak...

**Reply by gjsjohnmurray** [reply]
Maybe https://github.com/microsoft/vscode/pull/173885 is of interest to you. It's apparently already available in Insiders.

**Reply by msorens** [reply]
Thanks! Looks useful for other reasons but, alas, does not seem to address my issue here.

**Comment by alefragnani** [comment, marked as answer]
Hi @msorens ,

The new February update (v1.76) has announced a proposed API called [Tooltips for QuickPickItems](https://code.visualstudio.com/updates/v1_76#_tooltips-for-quickpickitems) .

Maybe this would fit your needs

**Reply by msorens** [reply]
As long as they do something sensible with longer tooltips (the example they show is just a single word!), then yes, it will most likely address my needs. Thank you for bringing it to my attention! 🎉 
Marking this as the answer (and keeping my fingers crossed regarding long tooltip support).
